siege_utilities/geo/gazetteers/wikidata_gazetteer.py

```
@@ -0,0 +1,429 @@
+        # Building shells first lets us run the contains() check on the
+        # polygon shell rather than re-implementing point-in-polygon.
+        shells = [Polygon(ring) for ring in outer_rings]
+        holes_for: list[list[list[tuple[float, float]]]] = [[] for _ in shells]
+        for inner_ring in inner_rings:
+            hole_poly = Polygon(inner_ring)
+            # Smallest containing shell wins (handles nested outer rings).
+            container_idx = None
+            container_area = None
+            for idx, shell in enumerate(shells):
+                if shell.covers(hole_poly):
+                    if container_area is None or shell.area < container_area:
```

Ruff (0.15.12)

[warning] Use a single `if` statement instead of nested `if` statements

(SIM102)